Windows cURL extension doesn't appear to support SSL

This was found in testing https://github.com/drupal/cms-launcher on Windows: https://github.com/drupal/cms-launcher/issues/38

It was also seen in https://github.com/crazywhalecc/static-php-cli/issues/597#issuecomment-2646556730

Any advice? Should I just try updating static-php-cli to 2.5.0 (we're on 2.4.5 right now) and rebuilding?

After updating to static-php-cli 2.5.0, I am also seeing the build fail on Windows: https://github.com/drupal/cms-launcher/actions/runs/13946411456/job/39034570532

The crux of the problem seems to be cURL, and the linker not being able to resolve it:

```
interface.obj : error LNK2001: unresolved external symbol __imp_curl_mime_init
interface.obj : error LNK2001: unresolved external symbol __imp_curl_mime_free
interface.obj : error LNK2001: unresolved external symbol __imp_curl_mime_addpart
interface.obj : error LNK2001: unresolved external symbol __imp_curl_mime_name
interface.obj : error LNK2001: unresolved external symbol __imp_curl_mime_filename
interface.obj : error LNK2001: unresolved external symbol __imp_curl_mime_type
interface.obj : error LNK2001: unresolved external symbol __imp_curl_mime_data
interface.obj : error LNK2001: unresolved external symbol __imp_curl_mime_data_cb
interface.obj : error LNK2001: unresolved external symbol __imp_curl_easy_escape
interface.obj : error LNK2001: unresolved external symbol __imp_curl_easy_unescape
interface.obj : error LNK2001: unresolved external symbol __imp_curl_free
interface.obj : error LNK2001: unresolved external symbol __imp_curl_global_init
interface.obj : error LNK2001: unresolved external symbol __imp_curl_global_cleanup
interface.obj : error LNK2001: unresolved external symbol __imp_curl_slist_append
interface.obj : error LNK2001: unresolved external symbol __imp_curl_slist_free_all
interface.obj : error LNK2001: unresolved external symbol __imp_curl_version_info
interface.obj : error LNK2001: unresolved external symbol __imp_curl_easy_strerror
interface.obj : error LNK2001: unresolved external symbol __imp_curl_easy_pause
interface.obj : error LNK2001: unresolved external symbol __imp_curl_easy_init
interface.obj : error LNK2001: unresolved external symbol __imp_curl_easy_setopt
interface.obj : error LNK2001: unresolved external symbol __imp_curl_easy_perform
interface.obj : error LNK2001: unresolved external symbol __imp_curl_easy_cleanup
interface.obj : error LNK2001: unresolved external symbol __imp_curl_easy_getinfo
interface.obj : error LNK2001: unresolved external symbol __imp_curl_easy_duphandle
interface.obj : error LNK2001: unresolved external symbol __imp_curl_easy_reset
interface.obj : error LNK2001: unresolved external symbol __imp_curl_easy_upkeep
multi.obj : error LNK2001: unresolved external symbol __imp_curl_multi_init
multi.obj : error LNK2001: unresolved external symbol __imp_curl_multi_add_handle
multi.obj : error LNK2001: unresolved external symbol __imp_curl_multi_remove_handle
multi.obj : error LNK2001: unresolved external symbol __imp_curl_multi_wait
multi.obj : error LNK2001: unresolved external symbol __imp_curl_multi_perform
multi.obj : error LNK2001: unresolved external symbol __imp_curl_multi_cleanup
multi.obj : error LNK2001: unresolved external symbol __imp_curl_multi_info_read
multi.obj : error LNK2001: unresolved external symbol __imp_curl_multi_strerror
multi.obj : error LNK2001: unresolved external symbol __imp_curl_multi_setopt
multi.obj : error LNK2001: unresolved external symbol __imp_curl_pushheader_bynum
share.obj : error LNK2001: unresolved external symbol __imp_curl_share_init
share.obj : error LNK2001: unresolved external symbol __imp_curl_share_setopt
```

--- a/config/lib.json
+++ b/config/lib.json
@@ -78,7 +78,6 @@
             "zlib"
         ],
         "lib-depends-windows": [
-            "openssl",
             "zlib",
             "libssh2",
             "nghttp2"

--- a/src/SPC/ConsoleApplication.php
+++ b/src/SPC/ConsoleApplication.php
@@ -32,7 +32,7 @@ use Symfony\Component\Console\Application;
  */
 final class ConsoleApplication extends Application
 {
-    public const VERSION = '2.5.0';
+    public const VERSION = '2.5.1';
 
     public function __construct()
     {

--- a/src/SPC/builder/windows/library/curl.php
+++ b/src/SPC/builder/windows/library/curl.php
@@ -37,6 +37,8 @@ class curl extends WindowsLibraryBase
                 '-DBUILD_EXAMPLES=OFF ' . // disable examples
                 '-DUSE_LIBIDN2=OFF ' . // disable libidn2
                 '-DCURL_USE_LIBPSL=OFF ' . // disable libpsl
+                '-DCURL_USE_SCHANNEL=ON ' . // use Schannel instead of OpenSSL
+                '-DCURL_USE_OPENSSL=OFF ' . // disable openssl due to certificate issue
                 '-DCURL_ENABLE_SSL=ON ' .
                 '-DUSE_NGHTTP2=ON ' . // enable nghttp2
                 '-DCURL_USE_LIBSSH2=ON ' . // enable libssh2

--- a/src/globals/ext-tests/curl.php
+++ b/src/globals/ext-tests/curl.php
@@ -3,3 +3,16 @@
 declare(strict_types=1);
 
 assert(function_exists('curl_init'));
+assert(function_exists('curl_setopt'));
+assert(function_exists('curl_exec'));
+assert(function_exists('curl_close'));
+$curl_version = curl_version();
+if (stripos($curl_version['ssl_version'], 'schannel') !== false) {
+    $curl = curl_init();
+    curl_setopt($curl, CURLOPT_URL, 'https://example.com/');
+    curl_setopt($curl, CURLOPT_RETURNTRANSFER, 1);
+    curl_setopt($curl, CURLOPT_HEADER, 0);
+    $data = curl_exec($curl);
+    curl_close($curl);
+    assert($data !== false);
+}

--- a/src/globals/test-extensions.php
+++ b/src/globals/test-extensions.php
@@ -21,10 +21,10 @@ $test_php_version = [
 
 // test os (macos-13, macos-14, ubuntu-latest, windows-latest are available)
 $test_os = [
-    'macos-13',
-    'macos-14',
+    // 'macos-13',
+    // 'macos-14',
     'ubuntu-latest',
-    // 'windows-latest',
+    'windows-latest',
 ];
 
 // whether enable thread safe
@@ -41,20 +41,20 @@ $prefer_pre_built = false;
 // If you want to test your added extensions and libs, add below (comma separated, example `bcmath,openssl`).
 $extensions = match (PHP_OS_FAMILY) {
     'Linux', 'Darwin' => '',
-    'Windows' => 'bz2,ctype,curl,dom,filter,gd,iconv,mbstring,opcache,openssl,pdo,pdo_sqlite,phar,session,simplexml,sqlite3,tokenizer,xml,xmlwriter,yaml,zip,zlib',
+    'Windows' => 'mbstring,tokenizer,phar,curl,openssl',
 };
 
 // If you want to test lib-suggests feature with extension, add them below (comma separated, example `libwebp,libavif`).
 $with_libs = match (PHP_OS_FAMILY) {
-    'Linux', 'Darwin' => 'mimalloc',
-    'Windows' => 'libjpeg,libavif,freetype,libwebp',
+    'Linux', 'Darwin' => '',
+    'Windows' => '',
 };
 
 // Please change your test base combination. We recommend testing with `common`.
 // You can use `common`, `bulk`, `minimal` or `none`.
 // note: combination is only available for *nix platform. Windows must use `none` combination
 $base_combination = match (PHP_OS_FAMILY) {
-    'Linux', 'Darwin' => 'bulk',
+    'Linux', 'Darwin' => 'minimal',
     'Windows' => 'none',
 };
 

